boilerplate/app/app.tsx

```diff
@@ -59,10 +60,12 @@ function App() {
     <ToggleStorybook>
       <RootStoreProvider value={rootStore}>
         <SafeAreaProvider initialMetrics={initialWindowMetrics}>
-          <AppNavigator
-            initialState={initialNavigationState}
-            onStateChange={onNavigationStateChange}
-          />
+          <ErrorBoundary catchErrors={"always"}>
+            <AppNavigator
+              initialState={initialNavigationState}
+              onStateChange={onNavigationStateChange}
+            />
+          </ErrorBoundary>
```

Comment:
Suggestion, not required for this PR: I find that it's generally good practice to also wrap screens within a navigator with an error boundary, so that navigation can continue to work if a component only nested within a single screen fails. I think it's helpful to be able to recommend that in a best-practices template.

However, this is mostly useful in apps that have navigation menus/bars outside the screen components (making sure those continue to work), so maybe it won't be super clear why that's helpful in this particular layout. Curious for your thoughts @jamonholmgren. I could see an argument for a separate PR that introduces an example navigation menu and then wraps individual screens.

Reply:
Good idea! I like this.

Reply:
@lindboe I think we should get a fresh new design for Ignite's boilerplate and add either a drawer or tab bar, after which we could incorporate your suggestion.